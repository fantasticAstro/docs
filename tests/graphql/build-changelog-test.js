const { createChangelogEntry } = require('../../script/graphql/build-changelog')

describe('creating a changelog from old schema and new schema', () => {
  it('finds a diff of schema changes, upcoming changes, and preview changes', async () => {
    const oldSchemaString = `
    type Query {
      stableField: String
      removedField: Boolean
      argumentsField(
        removedRequiredArgument: Int!
        removedOptionalArgument: Int!
        argumentMadeRequired: Int
        argumentMadeOptional: Int!
      ): String
    }
    `

    const newSchemaString = `
    type Query {
      stableField: String
      argumentsField(
        argumentMadeRequired: Int!
        argumentMadeOptional: Int
      ): String
    }
    `


    const entry = await createChangelogEntry(oldSchemaString, newSchemaString)
    expect(entry).toMatchSnapshot()
  })

  it('returns null when there isnt any difference', async () => {
    const schemaString = `
    type Query {
      i: Int!
    }`

    const nullEntry = await createChangelogEntry(schemaString, schemaString)
    expect(nullEntry).toBeNull()
  })
})
