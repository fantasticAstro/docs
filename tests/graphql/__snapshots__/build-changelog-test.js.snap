// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating a changelog from old schema and new schema finds a diff of schema changes, upcoming changes, and preview changes 1`] = `
Object {
  "date": "2020-11-20",
  "previewChanges": Array [],
  "schemaChanges": Array [
    Object {
      "changes": Array [
        "Field \`removedField\` was removed from object type \`Query\`",
        "Type for argument \`argumentMadeRequired\` on field \`Query.argumentsField\` changed from \`Int\` to \`Int!\`",
        "Type for argument \`argumentMadeOptional\` on field \`Query.argumentsField\` changed from \`Int!\` to \`Int\`",
        "Argument \`removedRequiredArgument: Int!\` was removed from field \`Query.argumentsField\`",
        "Argument \`removedOptionalArgument: Int!\` was removed from field \`Query.argumentsField\`",
      ],
      "title": "The GraphQL schema includes these changes:",
    },
  ],
  "upcomingChanges": Array [],
}
`;
